app/actions/AuthenticatedActionRefiner.scala

```diff
@@ -17,44 +17,41 @@
 package actions
 
 import actionsmodel.AuthenticatedRequest
-import cats.syntax.eq._
 import com.google.inject.Inject
 import config.AppConfig
 import models.GGCredId
 import play.api.Logger
 import play.api.mvc.Results.Redirect
 import play.api.mvc.{ActionRefiner, MessagesControllerComponents, Request, Result}
 import requests.RequestSupport._
+import uk.gov.hmrc.auth.core.AuthProvider.GovernmentGateway
 import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
 import uk.gov.hmrc.auth.core.retrieve.~
-import uk.gov.hmrc.auth.core.{AuthorisationException, AuthorisedFunctions, NoActiveSession}
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthProviders, AuthorisationException, AuthorisedFunctions, NoActiveSession}
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class AuthenticatedActionRefiner @Inject() (
-    af:        AuthorisedFunctions,
-    appConfig: AppConfig,
-    cc:        MessagesControllerComponents
+    val authConnector: AuthConnector,
+    appConfig:         AppConfig,
+    cc:                MessagesControllerComponents
 )(
     implicit
     ec: ExecutionContext
-) extends ActionRefiner[Request, AuthenticatedRequest] {
+) extends ActionRefiner[Request, AuthenticatedRequest] with AuthorisedFunctions {
```

Comment:
It was intentionally coded that way to promote composition over inheritance. It seemed to me more clear.

```
af.authorised()
```

as opposed to 
```
authorised()
``` 
clearly says where the function comes from.

More over it's not obvious what is the role of `val authConnector: AuthConnector,` on the constructor list.

Reply:
ah ok, it just requires a bit more set up in binding `AuthorisedFunctions` in a module. This way was suggested in during our PRA - it seems cleaner at least with this platform standard